specification/sbi_atee.adoc

```diff
@@ -13,6 +44,936 @@ running a VMM).
 The following operations must be implemented by the TSM ( _Formal input 
 parameters and return codes are TBD_ ).
 
+== Background
+
+Virtualization platforms are typically comprised of several components including
+platform firmware, host OS, VMM, and the actual payloads that run on them (typically
+in a VM). This model is well established, but the downside is that most platform
+components are in the TCB. This aspect is ill-suited for confidential compute
+workloads (Trusted Execution Environments) that strive to minimize the TCB footprint.
+
+We seek to alleviate this concern by introducing the notion of the a trusted broker for
+TEEs known as the TEE Security Manger (TSM). The idea is that the TSM has a minimal possible
+footprint, and acts as a trusted intermediary between the the untrusted platform components,
+and the TEE workloads that run on it. This is accomplished by having a clear seperation between
+the non-confidential and confidential portions of the workloads running on the platform. The exact
+hardware and software mechanisms used to guarantee the isolation are platform dependent, but
+confidential workloads have absolute execution, data integrity, and confidentiality protections
+from amy component in the non-confidential world. The responsibility of the TSM is to enforce 
+the guarantees accorded to TEE workloads. Given its minimal footprint, most of the functionality
+for actually running the payloads, like memory allocation, scheduling, interrupt handling, 
+virtual devices is left to the OS/VMM (which are outside the TCB).
```

Comment:
suggest add - "Explicitly, security invariants for confidential workloads are enforced by the TSM, as opposed to the VMM which continues to manage security for non-confidential workloads. On the other hand, resource management functions for all workloads (confidential and non-confidential) are managed by the untrusted OS/VMM.".